book/src/ephemeral-local-volumes.md

```diff
@@ -34,25 +34,89 @@ spec:
 
 ## Implementing inline ephemeral support
 
-Drivers must be modified (or implemented specifically) to support inline ephemeral workflows.  When Kubernetes encounters an inline CSI volume embedded in a pod spec, it treats that volume differently.  Mainly, the driver will only receive `NodePublish`, during the volume's mount phase, and `NodeUnpublish` when the pod is going away and the volume is unmounted.   To support inline, a driver must implement the followings:
+Drivers must be modified (or implemented specifically) to support inline
+ephemeral workflows. When Kubernetes encounters an inline CSI volume embedded
+in a pod spec, it treats that volume differently. Mainly, the driver will only
+receive `NodePublish`, during the volume's mount phase, and `NodeUnpublish` when
+the pod is going away and the volume is unmounted.
+
+Due to these requirements, ephemeral volumes will not be created using the [Controller
+Service](https://github.com/container-storage-interface/spec/blob/master/spec.md#controller-service-rpc),
+but the [Node
+Service](https://github.com/container-storage-interface/spec/blob/master/spec.md#node-service-rpc),
+instead. When the
+[kubelet](https://github.com/kubernetes/kubernetes/blob/70132b0f130acc0bed193d9ba59dd186f0e634cf/pkg/volume/csi/csi_mounter.go#L329)
+calls _NodePublishVolume_, it is the responsibility of the CSI driver to create the
+volume during that call, then publish the volume to the specified location. When
+the `kubelet` calls _NodeUnpublishVolume_, it is the responsibility of the CSI
+driver to delete the volume.
+
+To support inline, a driver must implement the followings:
 
 - Identity service
 - Node service
 
+### CSI Extension Specification
+
+#### [NodePublishVolume](https://github.com/container-storage-interface/spec/blob/master/spec.md#nodepublishvolume)
+
+##### Arguments
+
+* **`volume_id`**: Volume ID will be created by the Kubernetes and passed to the
+  driver by the kubelet.
+* **`volume_context["csi.storage.k8s.io/ephemeral"]`**: This value will be
```

Comment:
I would add here instead of the next section that the inline volume parameters were also copied into `volume_context`.

`fsType` is available via `NodePublishVolumeRequest.volume_capability.mount.fs_type`.